### PR TITLE
[#155688663] New Nav Sidebar

### PR DESCRIPTION
**Pull request by thematthopkins**
Adds components required by the new nav sidebar.  We'll hold off on QA until we get the full new nav release ready.

### CODE REVIEW

File: tests/integration/components/material-icon-test.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  integration: true
+});
+
+test('it renders', function(assert) {
+  // Set any properties with this.set('myProperty', 'value');
+  // Handle any actions with this.on('myAction', function(val) { ... });
+
+  this.render(hbs`{{material-icon icon="myIcon"}}`);
+
+  assert.equal(this.$().text().trim(), 'myIcon');
```
Comment (jonpitch) [inline tests/integration/components/material-icon-test.js]: @thematthopkins can we remove this jquery reference?

Comment (thematthopkins) [reply tests/integration/components/material-icon-test.js]: Good catch.  Done.

File: addon/components/b-panel.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    section.addEventListener("transitionend", transitionCompleted, false);
+    section.addEventListener("webkitTransitionEnd", transitionCompleted, false);
+
+    if(!get(this, 'open')){
+        section.style.height='0px';
+    }
+  },
+
+  openChanged: observer('open', function() {
```
Comment (jonpitch) [inline addon/components/b-panel.js]: just noticed this, if you wouldn't mind, we'll want to get rid of this and add an eslint rule to throw an exception. Maybe this is part of the side-effects we're seeing with the nav.

Comment (jonpitch) [reply addon/components/b-panel.js]: @thematthopkins you can disregard; I addressed in a different branch [here](https://github.com/b-lab-org/ember-cli-impact-core/commit/a706553f05598710b1bf63d58b4ec0396eee58bc).

File: addon/styles/impact/components/overflow-list.scss [hunk 3]
```diff
@@ -29,18 +29,26 @@ overflow-list{
...
+          padding-bottom: $impact-keyline-y-small/2;
+        }
 
-    a {
-      color: $impact-default-font-color;
-      display: block;
+        a {
+          color: $impact-default-font-color;
+        }
+      }
```
[the remark this answers is not in the document]
Comment (thematthopkins) [reply addon/styles/impact/components/overflow-list.scss]: Thanks @kikemoto !  Fixed.

**Pull request merged**